### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by HStakhiv**
image
image
image
image
image
image
image

### CODE REVIEW

File: taxi/tests/test_forms.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    def test_first_3_characters_should_be_uppercase(self):
+        self.assertFalse(self.form_data("t!s12345").is_valid())
+        self.assertFalse(self.form_data("Tes12345").is_valid())
+        self.assertFalse(self.form_data("TeS12345").is_valid())
+        self.assertFalse(self.form_data("TT012345").is_valid())
+
+    def test_last_5_characters_should_be_digits(self):
+        self.assertFalse(self.form_data("ASDF2345").is_valid())
+        self.assertFalse(self.form_data("ASD123#5").is_valid())
```
Comment (VadymShkarbul) [inline taxi/tests/test_forms.py:76]: We've seen very nice desisions with loop on today lection. And not nessesarly (it is my opinion) to check so many options. For example for 8 characters test: less, more, 8 and ок, check empty string

**Review by VadymShkarbul — CHANGES_REQUESTED** [review]

File: taxi/tests/test_admin.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    # Does not work :(
+
+    # def test_driver_add_license_number_listed(self):
+    #     """Test that driver's license_number
+    #     is on driver add admin page"""
+    #
+    #     url = reverse("admin:taxi_driver_add")
+    #     result = self.client.get(url)
+    #
+    #     self.assertContains(result, self.driver.license_number)
```
[the remark this answers is not in the document]
Comment (HStakhiv) [reply taxi/tests/test_admin.py]: I found a solution )

File: taxi/tests/test_forms.py (hunk 1 above)
Comment (HStakhiv) [reply taxi/tests/test_forms.py:76]: Yes, you are right

File: taxi/tests/test_admin.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (VadymShkarbul) [reply taxi/tests/test_admin.py]: Yes. I saw it later. A bit get lost in commits ;)

**Review by VadymShkarbul — APPROVED** [review]

**Review by Nattalli — CHANGES_REQUESTED** [review]
Please, update your version with the master's one. Add tests for the searching functionality. Add tests with `post-requests`

**Comment by Nattalli** [comment]
@HStakhiv my bad, I noticed that you already have tests for post-requests. So just update your version with the master's one and add tests for the searching functionality.

File: taxi/tests/test_views.py [hunk 3]
```diff
@@ -41,6 +41,15 @@ def test_retrieve_manufacturer(self):
         )
         self.assertTemplateUsed(response, "taxi/manufacturer_list.html")
 
+    def test_manufacturer_search_test(self):
+        response = self.client.get(MANUFACTURER_LIST_URL + "?name=0")
+        response_res = list(response.context["manufacturer_list"])
```
Comment (Y-Havryliv) [inline taxi/tests/test_views.py:46]: did you mean result?)

File: taxi/tests/test_views.py [hunk 4]
```diff
@@ -41,6 +41,15 @@ def test_retrieve_manufacturer(self):
...
+        manufacturers_res = list(
```
Comment (Y-Havryliv) [inline taxi/tests/test_views.py:47]: same here, `result` is not that long, right?)

File: taxi/tests/test_views.py [hunk 5]
```diff
@@ -83,6 +92,13 @@ def test_car_detail_retrieve(self):
 
         self.assertEqual(response.status_code, 200)
 
+    def test_car_search_test(self):
+        response = self.client.get(CAR_LIST_URL + "?model=e")
+        response_res = list(response.context["car_list"])
```
Comment (Y-Havryliv) [inline taxi/tests/test_views.py:97]: please, fix everywhere

**Review by Y-Havryliv — CHANGES_REQUESTED** [review]

File: taxi/tests/test_views.py (hunk 5 above)
Comment (HStakhiv) [reply taxi/tests/test_views.py]: I choose a different path. Can it be so?

**Review by Y-Havryliv — APPROVED** [review]
GJ!

**Pull request closed without merging**